src/components/main.tsx

```diff
@@ -1,29 +1,59 @@
 import React, { useState, useEffect } from "react";
 import BrightEthereumDeepLinkQR from "./qrGenerator";
-import { Button, Form, Input, Modal, ModalBody, Container, InputGroup, InputGroupAddon } from 'reactstrap';
+import { Button, Form, Input, Modal, ModalBody, Container, InputGroup, InputGroupAddon, FormFeedback } from 'reactstrap';
 import { assignEthereum } from '../util/metamask'
+import { convertENS } from '../util/ens'
 import MetaMask from '../assets/img/MetaMask.svg'
 import Web3 from 'web3';
 import { mobileCheck, androidOrIphoneLink } from "util/detectMobile";
 import { deepLinkPrefix } from "util/deepLink";
 
 const Main = () => {
     const [input, updateInput] = useState('');
+    const [address, updateAddress] = useState('');
     const [showQR, toggleShowQR] = useState(false);
     const [notSupported, toggleNotSupported] = useState(false);
-    const [validAddress, isValid] = useState(false);
+    const [invalidError, setInvalidError] = useState(false);
+    const [validAddress, isValid] = useState(true);
     const [isMobile] = useState(mobileCheck())
     const [ethereum] = useState(() => assignEthereum());
     const [brightIdStoreLink] = useState(androidOrIphoneLink());
 
-    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const handleChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
         updateInput(e.target.value)
+        setInvalidError(false)
     };
 
     useEffect(() => {
-        isValid(Web3.utils.isAddress(input));
+        if (input.includes('.eth')) {
+            async function resolveENS() {
+                validateAndUpdateAddress(await convertENS(input))
+            }
+            resolveENS();
+         } else validateAndUpdateAddress(input);
     }, [input])
 
+    const validateAndUpdateAddress = (input: string) => {
+        if ( Web3.utils.isAddress(input) ) {
+            isValid(true);
+            updateAddress(input);
+        } else {
+            isValid(false);
+        }
+    }
+
+    const submitAddress = () => {
+
+      if(validAddress)
+        {
+          toggleShowQR(true);
+          setInvalidError(false)
```

Comment:
This is not necessary since you are already calling `setInvalidError(false)` on every input change.
However, to get invalidError to update properly in the DOM, you should set inside of the `useEffect` not in `handleChange`